_posts/2024-03-28-fuse-storage-driver.adoc

```diff
@@ -0,0 +1,84 @@
+
+All tests presented in Fig. 2. were run on an OpenShift Dedicated 4.15.3 cluster with a productized version of Eclipse Che 7.82.1. 
+
+[cols="1,1"]
+|===
+|Test name |Project URL 
+
+|dashboard
+|https://github.com/eclipse-che/che-dashboard[che-dashboard]
+
+|dashboard-edit
+|https://github.com/eclipse-che/che-dashboard[che-dashboard]
+
+|operator
+|https://github.com/eclipse-che/che-operator[che-operator]
+
+|operator-edit
+|https://github.com/eclipse-che/che-operator[che-operator]
+
+|server
+|https://github.com/eclipse-che/che-server[che-server]
+
+|server-edit
+|https://github.com/eclipse-che/che-server[che-server]
+
+|nginx-alpine
+|https://github.com/nginxinc/docker-nginx/tree/master/stable/alpine[docker-nginx]
+
+|quarkus-api-example
+|https://github.com/che-incubator/quarkus-api-example[quarkus-api-example]
+|===
+
+Fig. 2. Table of test names name project URLs.
+
+For each test apart from the <projectname>-edit tests, the `podman system reset` command was run before running each test in order to clear the graphroot directory. This removes all image layers, therefore these tests measure a clean-build scenario, just like when a user creates and runs `podman build` in a new CDE.
+
+For the <projectname>-edit tests, the image container was built beforehand. The tests measure a rebuild of the image after changing the source code without running `podman system reset` command. This scenario mimics the case where the developer is running a new build after making code changes in their CDE.
+
+image::/assets/img/fuse-storage-driver/image-build-time.png[]
+Fig. 3. Container image build time results
+
+image::/assets/img/fuse-storage-driver/graphroot-size.png[]
+Fig. 4. Graphroot directory size results
+
+For all tests, the fuse-overlayfs storage driver had faster build times and significantly smaller storage consumption compared to vfs. The benefits of CoW is especially evident in Fig. 4. For example the operator-edit test showed that by using fuse-overlayfs, the graphroot directory was about 88% smaller, saving about 15GB of storage.
+
+## Conclusion
+
+In conclusion, the fuse-overlayfs storage driver should be considered for your Eclipse Che CDEs, saving time and storage thanks to the CoW support. In general, there is no “best” storage driver because performance, stability and usability of each storage driver is dependent on your specific workload and environment. However with vfs not supporting CoW, fuse-overlayfs is a much more suitable choice for image layer management.
```

Comment:
```suggestion
In conclusion, the fuse-overlayfs storage driver should be considered for your Eclipse Che CDEs, saving time and storage thanks to the CoW support. In general, there is no “best” storage driver because performance, stability, and usability of each storage driver is dependent on your specific workload and environment. However, with vfs not supporting CoW, fuse-overlayfs is a much more suitable choice for image layer management.
```